iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -407,32 +407,53 @@ static void populateTilingToInvocationPatterns(
   auto getThreadProcInfoFn = [&launchConfig](
                                  OpBuilder &builder, Location loc,
                                  ArrayRef<Range> parallelLoopRanges) {
+    int numLevels = parallelLoopRanges.size();
     Type indexType = builder.getIndexType();
-    SmallVector<linalg::ProcInfo, 2> procInfo(2);
-    procInfo[1] = {builder.create<gpu::ThreadIdOp>(loc, indexType,
-                                                   builder.getStringAttr("x")),
-                   builder.create<ConstantIndexOp>(
-                       loc, launchConfig.getWorkgroupSize()[0])};
-    procInfo[0] = {builder.create<gpu::ThreadIdOp>(loc, indexType,
-                                                   builder.getStringAttr("y")),
-                   builder.create<ConstantIndexOp>(
-                       loc, launchConfig.getWorkgroupSize()[1])};
+    SmallVector<linalg::ProcInfo, 4> procInfo(numLevels);
+    if (numLevels > 0) {
+      procInfo[--numLevels] = {builder.create<gpu::ThreadIdOp>(
```

Comment:
This is super confusing. 

Can we just use getGPUProcessorIdsAndCounts? THe only difference seems to be that the number of processors is obtained from WorkgroupSize, instead of being from `gpu::BlockDimOp`. But with the distribution you are using that should not be needed.